scvi/dataset/dataset.py

```diff
@@ -577,13 +577,14 @@ def initialize_cell_attribute(self, attribute_name, attribute, categorical=False
                 )
             )
             attribute_name = valid_attribute_name
-        if not self.nb_cells == len(attribute):
+        len_attribute = attribute.shape[0] if type(attribute) is not list else len(attribute)
```

Comment:
yes, `len` is not defined for sparse matrices.

Reply:
can you use duck-typing instead of looking for type equality, which is somewhat restrictive.
```python
try:
   len_attribute = attribute.shape[0]
except AttributeError:
   len_attribute = len(attribute)
```

Reply:
pushed.